### ISSUE TITLE
.NET Core April Update - 2.1.10 and 2.2.4

### ISSUE THREAD

**Issue by leecow** [dotnet/core#2555]
Included fixes address security issues detailed in the release notes.

## .NET Core

If you are a Visual Studio customer, please read the note included in the Release Notes regarding which SDK is supported by your version of Visual Studio. 

* **.NET Core 2.1.10 and .NET Core SDK 2.1.506**
  [Release Notes](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.10/2.1.10.md)
  [Download 2.1](https://dotnet.microsoft.com/download/dotnet-core/2.1)

* **.NET Core 2.2.4 and .NET Core SDK 2.2.106**
  [Release Notes](https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2.4/2.2.4.md)
  [Download 2.2](https://dotnet.microsoft.com/download/dotnet-core/2.2)

Please report any issues you find with these releases, either responding to this issue, creating a new issue here or creating a new issue in one of the following repos:

* [aspnet/Home](https://github.com/aspnet/Home/issues) - for ASP.NET Core questions
* [aspnet/EntityFrameworkCore](https://github.com/aspnet/EntityFrameworkCore/issues) - for Entity Framework Core questions
* [dotnet/cli](https://github.com/dotnet/cli/issues) - for CLI tools and questions
* [dotnet/corefx](https://github.com/dotnet/corefx/issues) - for API issues and questions
* [dotnet/coreclr](https://github.com/dotnet/coreclr/issues) - for runtime issues
* [dotnet/core-setup](https://github.com/dotnet/core-setup/issues) - for installer issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

## .NET Support Policies

Microsoft support policies are defined in the following documents:

* [Microsoft .NET Support Policy](https://www.microsoft.com/net/support/policy)
* [.NET Core OS Lifecycle Policy](https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md)

**Comment by Nothing-Works** [comment]
@leecow  Hi I am Mac user, just went to the [download page](https://dotnet.microsoft.com/download/dotnet-core/2.2) and I am wondering why mac does not have 2.2.203 only 2.2.106?

I am quite new to .Net Core, does it mean that as long as it's 2.2.4, the SDK version does not matter?

**Comment by fschlaef** [comment]
@leecow be advised, the "Blog Post" link in https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2.4/2.2.4.md#changes-in-224 forwards to 

https://devblogs.microsoft.com/dotnet/net-core-march-2019/

Should be :

https://devblogs.microsoft.com/dotnet/net-core-april-2019-updates-2-1-10-and-2-2-4/

**Comment by karelz** [comment]
@fschlaef thanks! Fixed: https://github.com/dotnet/core/commit/1b3b626848613320599f945d70a8583dff3ccac1#diff-3fe8fb440ad10bde0667d99db10fa9ce ... in future, feel free to submit PR with the fix ;)

**Comment by mattzink** [comment]
Which SDK do I install if I used both VS 2017 and 2019?

**Comment by leecow** [comment]
@mattzink - We're going to add some decorations to the download site to help with that information. In the meantime, details are in the release notes. Short answer: 2.1.506 or 2.2.106 for 2017 and 2.1.603 or 2.2.203 for 2019.

**Comment by mattzink** [comment]
@leecow Sorry, I probably wasn't clear. I need to use **both** VS 2017 and 2019 depending on the project. Should I install both 2.2.106 and 2.2.203 SDKs?

**Comment by fschlaef** [comment]
I'll add another question : I accidentally downloaded and installed 2.2.203 but I use Visual Studio 2017, however everything seems to be working just fine. 

Can I keep it like this or should I install 2.2.106 instead ?
What are the differences between the two versions ?

**Comment by nprokopic** [comment]
@fschlaef It seemed to be working for me as well on my dev machine (version `2.2.202`), but on Team City I've got MSBuild compatibility error like [this](https://stackoverflow.com/questions/55420731/msbuild-fails-for-net-core-sdk-version-2-2-202):

> Version 2.2.202 of the .NET Core SDK requires at least version 16.0.0 of MSBuild. The current available version of MSBuild is 15.6.82.30579. Change the .NET Core SDK specified in global.json to an older version that requires the MSBuild version currently available.

Then found this `2.2.1XX` (for VS 2017) vs `2.2.2XX` (for VS 2019) difference and went with `2.2.1XX` train.

**Comment by leecow** [comment]
@mattzink  and @nprokopic - It looks like VS on Windows has a bit of a safety buffer for this issue in most cases though, as pointed out, there are some scenarios where MSBuild compat will come up. 

@fschlaef - the key differences are lower level tooling updates; compilers, MSBuild and NuGet being the key components.

**Comment by dhuyvett** [comment]
The download link for aspnetcore on linux-x86 is broken
`https://dotnet.microsoft.com/download/dotnet-core/2.1` ->
`https://dotnet.microsoft.com/download/thank-you/dotnet-runtime-2.1.10-linux-x64-asp.net-core-runtime-binaries` ->
`https://download.visualstudio.microsoft.com/download/pr/5967af9f-21d4-4b2f-a166-52af457d02aa/713e9a0f3753cf79171594bbdb92ad89/aspnetcore-runtime-2.1.10-linux-x64.tar.gz` ->
Runtime Error

**Comment by leecow** [comment]
@dhuyvett - I just tried all of the 2.1 asp links and things seem to be working ok now. Try again?

**Comment by dhuyvett** [comment]
@leecow : confirmed that it is working now.

**Comment by matthid** [comment]
It seems like this is not yet available on the ubuntu feed:

```
root@0b03e44d6b70:/# apt-cache madison dotnet-sdk-2.1
dotnet-sdk-2.1 |  2.1.603-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.602-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.505-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.504-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.503-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.502-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.500-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.403-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.402-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.401-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.400-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.302-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.301-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
dotnet-sdk-2.1 |  2.1.300-1 | https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages
```

Is there always a delay? Is there a `.deb` available somewhere to install it manually?

**Comment by leecow** [comment]
@matthid - this was a communication error on our part. The ever growing list of versions is making the story very hard to tell. The latest SDKs released for Linux are 2.1.603 and 2.2.203.

@vivmishra @jamshedd

**Comment by karelz** [comment]
Fully replaced by #2727
